internal/schema/v1/search_history.go

```diff
@@ -0,0 +1,30 @@
+package schema
+
+import "github.com/Decentr-net/cerberus/internal/schema/types"
+
+// SearchHistory is user's search history.
+type SearchHistory struct {
+	Timestamp
+
+	Engine     string `json:"searchEngine"`
+	SearchLine string `json:"searchLine"`
+}
+
+// Type ...
+func (SearchHistory) Type() types.Type {
+	return types.PDVSearchHistoryType
+}
+
+// Validate ...
+func (d SearchHistory) Validate() bool {
+	if d.Engine == "" || d.SearchLine == "" {
```

Comment:
Is there any need for length validation?

Reply:
done